### ISSUE TITLE
opening several files at once: different sequence

### ISSUE THREAD

**Issue by Golddouble** [cboxdoerfer/fsearch#363]
I have found a list of txt-files in FSearch.
Now I want to open 4 of this files at once.

I select this files and right click "open with Featherpad".

My expectation was, that the sequence of the tabs in Featherpad is the same as the sequence in FSearch. But it is different:
![k20220614-102413](https://user-images.githubusercontent.com/13977359/173530289-a7add068-1efe-47c5-9c65-60d8554a13c0.jpg)

As you can see in Fsearch the sequece is a-b-c-d (see the yellowish letter).
But in Featherpad it is a-c-b-e.

I would like to have the same sequence.

Question:
What's the rule in which sequence FSearch opens the files?

Thank you.

**Comment by cboxdoerfer** [comment]
Thanks for the report.

> What's the rule in which sequence FSearch opens the files?

It's currently undefined. The selection is stored in what's called a hash map and iterating over the selected items in that map has no defined order, so it can produce different results every time the selection changes.

But it's possible to use a different way to iterate over all items, following the order from the result list. This likely comes at a performance cost, but I don't think this will be noticeable, so I'll fix that with the next release.

**Comment by Golddouble** [comment]
Thank you.

> This likely comes at a performance cost, but I don't think this will be noticeable, so I'll fix that with the next release.
You mean, that 

You mean that it then will take a bit longer to open all files in Featherpad? Do I understand that correctly? If so, that is not a problem.

> But it's possible to use a different way to iterate over all items, following the order from the result list.

Thank you. Would be great, if the current sort order in FSearch is the order in which the files are opened. So when I sort against file name, it should take this order for opening. And if it is sorted according date, it should take this sort order.

> so I'll fix that with the next release.

Thank you very much.

**Comment by cboxdoerfer** [comment]
> You mean that it then will take a bit longer to open all files in Featherpad? Do I understand that correctly? If so, that is not a problem.

Yes, but I guess it's only going to be a few milliseconds with millions of results. So really not an issue.

> Thank you. Would be great, if the current sort order in FSearch is the order in which the files are opened. So when I sort against file name, it should take this order for opening. And if it is sorted according date, it should take this sort order.

Yes, that's really how it should work. Everything else makes little sense imho.
